src/tests/pmfs_demo.c

```diff
@@ -0,0 +1,1526 @@
+		pmfs_combine_pool_fini_list(&g_test_fini_list);
+	}
+
+	return *pmfs_pool;
+}
+
+static struct pmfs_context *
+pmfs_set_ctx(void)
+{
+	struct pmfs_context *pmfs_ctx;
+
+	D_ALLOC(pmfs_ctx, sizeof(struct pmfs_context));
+	D_ASSERT(pmfs_ctx != NULL);
+
+	D_INIT_LIST_HEAD(&pmfs_ctx->pmfs_pool.pl);
+	/* add pool mapping to /mnt/daos/pmfs_cli0.pmem, 8G NVME,2G SCM, skip container create */
+	pmfs_ctx->pmfs_pool = pmfs_add_single_pool("/mnt/daos/pmfs_cli0.pmem",
+						 8ULL << 30, 2ULL << 30, true, false);
+	/* That aims to associated with engine lib */
+	pmfs_ctx->pmfs_pool.pl = g_test_pool_list;
+	pmfs_ctx_combine_pool_list(pmfs_ctx);
+	return pmfs_ctx;
+}
+
+static void
+pmfs_mkfs_cb(void *arg)
+{
+	struct mkfs_args *mags = arg;
+
+	pmfs_mkfs(mags->poh, mags->uuid);
+}
+
+static void
+pmfs_mount_cb(void *arg)
+{
+	struct mount_args *mount_args = arg;
+
+	pmfs_mount(mount_args->poh, mount_args->coh, mount_args->flags,
+			mount_args->pmfs);
+}
+
+static void
+pmfs_umount_cb(void *arg)
+{
+	struct umount_args *umount_args = arg;
+
+	pmfs_umount(umount_args->pmfs);
+}
+
+static void
+pmfs_mkdir_cb(void *arg)
+{
+	struct mkdir_args *mkdir_args = arg;
+
+	pmfs_mkdir(mkdir_args->pmfs, mkdir_args->parent, mkdir_args->name,
+			mkdir_args->mode);
+}
+
+static void
+pmfs_listdir_cb(void *arg)
+{
+	struct listdir_args *listdir_args = arg;
+
+	pmfs_listdir(listdir_args->pmfs, listdir_args->obj, &listdir_args->nr);
+}
+
+static void
+pmfs_remove_cb(void *arg)
+{
+	struct remove_args *remove_args = arg;
+
+	pmfs_remove(remove_args->pmfs, remove_args->parent, remove_args->name,
+			remove_args->force, remove_args->oid);
+}
+
+static void
+pmfs_open_cb(void *arg)
+{
+	struct open_args *open_args = arg;
+
+	pmfs_open(open_args->pmfs, open_args->parent, open_args->name,
+			open_args->mode, open_args->flags, open_args->chunk_size,
+			open_args->value, &open_args->obj);
+}
+
+static void
+pmfs_readdir_cb(void *arg)
+{
+	struct readdir_args *readdir_args = arg;
+
+	pmfs_readdir(readdir_args->pmfs, readdir_args->obj, readdir_args->nr,
+			readdir_args->dirs);
+}
+
+static void
+pmfs_lookup_cb(void *arg)
+{
+	struct lookup_args *lookup_args = arg;
+
+	pmfs_lookup(lookup_args->pmfs, lookup_args->path, lookup_args->flags,
+			&lookup_args->obj, lookup_args->mode, lookup_args->stbuf);
+}
+
+static void
+pmfs_release_cb(void *arg)
+{
+	struct release_args *release_args = arg;
+
+	pmfs_release(release_args->obj);
+}
+
+static void
+pmfs_punch_cb(void *arg)
+{
+	struct punch_args *punch_args = arg;
+
+	pmfs_punch(punch_args->pmfs, punch_args->obj, punch_args->offset,
+			punch_args->len);
+}
+
+static int
+pmfs_write_cb(void *arg)
+{
+	struct write_args *write_args = arg;
+	d_sg_list_t     *sgl = write_args->user_sgl;
+	daos_size_t     off = write_args->off;
+	int     rc = 0;
+
+	rc = pmfs_write_sync(write_args->pmfs, write_args->obj, sgl, off);
+	if (rc != 0) {
+		D_PRINT("write error\r\n");
+		return rc;
+	}
+
+ 	return 0;
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs